lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -982,7 +958,51 @@ class SemanticsObject {
   final EngineSemanticsOwner owner;
 
   /// The DOM element used to convey semantics information to the browser.
-  final DomElement element = domDocument.createElement('flt-semantics');
+  late DomElement element;
+
+  void _updateElement(PrimaryRole role) {
+    switch (role) {
+      case PrimaryRole.link:
+        element = domDocument.createElement('a');
```

Comment:
The `element` field is designed to be `final`, as it participates in the parent's DOM structure. Simply swapping the element field will result in a dangling element not attached to anywhere. Let's chat about what you are trying to do and find a better solution.